test/blocks/merch/merch.test.js

```diff
@@ -17,6 +17,32 @@ const config = {
   env: { name: 'prod' },
 };
 
+/**
+ * utility function that tests Price spans against mock HTML
+ *
+ * @param {util} selector price span selector
+ * @param {*} expectedAttributes { <attribute key in element dataset>:
+ * <expected attribute value, UNDEF if should be undefined>}
+ */
+const validatePriceSpan = async (selector, expectedAttributes) => {
+  const el = await merch(document.querySelector(
+    selector,
+  ));
+  const { nodeName, dataset } = await el.onceSettled();
+  expect(nodeName).to.equal('SPAN');
+  if (!expectedAttributes.template) {
+    expect(dataset.template).to.be.undefined;
+  }
+  Object.keys(expectedAttributes).forEach((key) => {
+    const value = expectedAttributes[key];
+    if (value === 'UNDEF') {
+      expect(dataset[key]).to.be.undefined;
+    } else {
+      expect(dataset[key]).to.equal(value);
+    }
+  });
```

Comment:
The `.equal` Chai assertion should be able to work with the regular `undefined` value as well, per https://www.chaijs.com/api/bdd/#method_equal.

Based on this, the `'UNDEF'` strings do not seem to be necessary:
```
Object.entries(expectedAttributes).forEach(([key, value]) => {
      expect(dataset[key]).to.equal(value);
});
// validatePriceSpan('.merch.price.term', { displayRecurrence: undefined });
```